ghc-lib-gen/src/Ghclibgen.hs

```diff
@@ -446,33 +446,37 @@ applyPatchGhcPrim ghcFlavor = do
       writeFile tysPrim .
           replace
               "bcoPrimTyCon = pcPrimTyCon0 bcoPrimTyConName LiftedRep"
-              "bcoPrimTyCon = pcPrimTyCon0 bcoPrimTyConName UnliftedRep" .
+              "\n#if __GLASGOW_HASKELL >= 900\nbcoPrimTyCon = pcPrimTyCon0 bcoPrimTyConName LiftedRep\n#else\nbcoPrimTyCon = pcPrimTyCon0 bcoPrimTyConName UnliftedRep\n#endif\n" .
           replace
               "bcoPrimTyConName              = mkPrimTc (fsLit \"BCO\") bcoPrimTyConKey bcoPrimTyCon"
-              "bcoPrimTyConName              = mkPrimTc (fsLit \"BCO#\") bcoPrimTyConKey bcoPrimTyCon"
+              "\n#if __GLASGOW_HASKELL >= 900\nbcoPrimTyConName              = mkPrimTc (fsLit \"BCO\") bcoPrimTyConKey bcoPrimTyCon\n#else\nbcoPrimTyConName              = mkPrimTc (fsLit \"BCO#\") bcoPrimTyConKey bcoPrimTyCon\n#endif\n"
         =<< readFile' tysPrim
         )
+
     let createBCO = "libraries/ghci/GHCi/CreateBCO.hs"
     when (ghcFlavor >= Ghc901) (
       writeFile createBCO .
+          replace
+              "{-# LANGUAGE RecordWildCards #-}"
+              "{-# LANGUAGE RecordWildCards #-}\n{-# LANGUAGE CPP #-}" .
           replace
               "do linked_bco <- linkBCO' arr bco"
-              "do BCO bco# <- linkBCO' arr bco" .
+              "\n#if __GLASGOW_HASKELL__ >= 900\n     do linked_bco <- linkBCO' arr bco\n#else\n     do BCO bco# <- linkBCO' arr bco\n#endif" .
```

Comment:
shouldn’t this check on ghc-prim instead of GHC? Given how closely they are tied together, it probably doesn’t matter too much but it is a bit confusing.

Reply:
Issue https://github.com/digital-asset/ghc-lib/issues/275